implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicHttpClientProviderImpl.java

```diff
@@ -0,0 +1,296 @@
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.TimeGauge;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.config.MissingRequiredConfigurationException;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.util.*;
+import io.micrometer.core.ipc.http.HttpSender;
+import io.micrometer.core.ipc.http.HttpUrlConnectionSender;
+
+/**
+ * Publishes metrics to New Relic Insights REST API.
+ * 
+ * @author Jon Schneider
+ * @author Johnny Lim
+ * @author Neil Powell
+ */
+public class NewRelicHttpClientProviderImpl implements NewRelicClientProvider {
+
+    private final Logger logger = LoggerFactory.getLogger(NewRelicHttpClientProviderImpl.class);
+
+    private final NewRelicConfig config;
+    private final HttpSender httpClient;
+    private final String insightsEndpoint;
+    private final NamingConvention namingConvention;
+
+    @SuppressWarnings("deprecation")
+    public NewRelicHttpClientProviderImpl(NewRelicConfig config) {
+        this(config, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()), new NewRelicNamingConvention());
+    }
+
+    public NewRelicHttpClientProviderImpl(NewRelicConfig config, HttpSender httpClient, NamingConvention namingConvention) {
+
+        if (!config.meterNameEventTypeEnabled() && StringUtils.isEmpty(config.eventType())) {
+            throw new MissingRequiredConfigurationException("eventType must be set to report metrics to New Relic");
+        }
+        if (StringUtils.isEmpty(config.accountId())) {
+            throw new MissingRequiredConfigurationException("accountId must be set to report metrics to New Relic");
+        }
+        if (StringUtils.isEmpty(config.apiKey())) {
+            throw new MissingRequiredConfigurationException("apiKey must be set to report metrics to New Relic");
+        }
+        if (StringUtils.isEmpty(config.uri())) {
+            throw new MissingRequiredConfigurationException("uri must be set to report metrics to New Relic");
+        }
+
+        this.config = config;
+        this.httpClient = httpClient;
+        this.namingConvention = namingConvention;
+        this.insightsEndpoint = config.uri() + "/v1/accounts/" + config.accountId() + "/events";
+    }
+
+    @Override
+    public void publish(NewRelicMeterRegistry meterRegistry, List<Meter> meters) {
+        // New Relic's Insights API limits us to 1000 events per call
+        // 1:1 mapping between Micrometer meters and New Relic events
+        for (List<Meter> batch : MeterPartition.partition(meterRegistry, Math.min(config.batchSize(), 1000))) {
+            for (Meter meter : batch) {
+                sendEvents(meter.getId(),
```

Comment:
This looks like it'll make an HTTP call for each meter. Previously it made one HTTP call per batch, if my understanding is correct.

Reply:
Good catch. Reverted.